python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1173,9 +1173,13 @@ def build_grouped_ndrange_for(ctx, node):
                     f"Group for should have 1 loop target, found {len(targets)}"
                 )
             target = targets[0]
-            target_var = impl.expr_init(
-                matrix.Vector([0] * len(ndrange_var.dimensions),
-                              dt=primitive_types.i32))
+            if current_cfg().real_matrix:
+                mat = matrix.make_matrix([0] * len(ndrange_var.dimensions))
```

Comment:
Maybe good to pass a `dt` to make_matrix here, too.

Reply:
thanks！